core/src/main/java/org/elasticsearch/search/suggest/completion/CompletionSuggestion.java

```diff
@@ -54,6 +54,48 @@ public CompletionSuggestion(String name, int size) {
         super(name, size);
     }
 
+    private class OptionPriorityQueue extends org.apache.lucene.util.PriorityQueue<Entry.Option> {
+
+        public OptionPriorityQueue(int maxSize) {
+            super(maxSize);
+        }
+
+        @Override
+        protected boolean lessThan(Entry.Option a, Entry.Option b) {
+            return sortComparator().compare(a, b) > 0;
+        }
+
+        public Entry.Option[] get() {
+            int size = size();
+            Entry.Option[] results = new Entry.Option[size];
+            for (int i = size - 1; i >= 0; i--) {
+                results[i] = pop();
+            }
+            return results;
+        }
+    }
+
+    @Override
+    public Suggest.Suggestion<Entry> reduce(List<Suggest.Suggestion<Entry>> toReduce) {
+        if (toReduce.size() == 1) {
+            return toReduce.get(0);
+        } else {
+            OptionPriorityQueue priorityQueue = new OptionPriorityQueue(size);
```

Comment:
Maybe some docs explaining what `OptionPriorityQueue` is doing?  I.e. we are combining multiple results from N shards on the coordinating node right?